File: RadioControl/app/src/main/java/com/nikhilparanjape/radiocontrol/receivers/PersistenceAlarmReceiver.java
package com.nikhilparanjape.radiocontrol.receivers;

import android.content.Context;
import android.content.Intent;
import android.support.v4.content.WakefulBroadcastReceiver;

import com.nikhilparanjape.radiocontrol.services.PersistenceService;

/**
 * Created by admin on 7/9/2017.
 */

public class PersistenceAlarmReceiver extends WakefulBroadcastReceiver{
    public static final int REQUEST_CODE = 25912;

    @Override
    public void onReceive(Context context, Intent intent) {
        Intent service = new Intent(context, PersistenceService.class);
        startWakefulService(context, service);
    }
}

File: RadioControl/app/src/main/java/com/nikhilparanjape/radiocontrol/receivers/PersistenceAlarmReceiver.kt
@file:Suppress("DEPRECATION")

package com.nikhilparanjape.radiocontrol.receivers

import android.content.Context
import android.content.Intent
import androidx.legacy.content.WakefulBroadcastReceiver

import com.nikhilparanjape.radiocontrol.services.PersistenceService

/**
 * Created by admin on 7/9/2017.
 */

class PersistenceAlarmReceiver : WakefulBroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        val service = Intent(context, PersistenceService::class.java)
        WakefulBroadcastReceiver.startWakefulService(context, service)
    }
}
